core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -167,16 +172,25 @@ Partition get(StructLike key) {
     private final StructLike key;
     private long recordCount;
     private int fileCount;
+    private int deleteFileCount;
 
     Partition(StructLike key) {
       this.key = key;
       this.recordCount = 0;
       this.fileCount = 0;
+      this.deleteFileCount = 0;
     }
 
     void update(DataFile file) {
       this.recordCount += file.recordCount();
       this.fileCount += 1;
     }
+
+    void updateDeleteFile(List<DeleteFile> deleteFiles) {
+      for (DeleteFile deleteFile : deleteFiles) {
+        this.recordCount += deleteFile.recordCount();
```

Comment:
This should not be the same record count. Data record counts are tracked separately from both equality delete count and position delete count. This should actually add 3 columns: delete_file_count, equality_delete_count, and position_delete_count.

Reply:
> Thanks, @coolderli! This is going to be really helpful. Another thing that we should do is add a new metadata table that loads just delete files. Are you interested in working on that as well?

Yes, I'll do that. I will fix this and push another pr for that later.

Reply:
Thanks, I'll fix it.